source/css/sass/_mixins--spacing.scss

```diff
@@ -1,44 +1,156 @@
+    padding-left: #{get-rem-from-px($size_in_px)}rem;
+  }
+}
+
+@mixin _padding-right($size-in-px) {
+  @if $size-in-px == 0 {
+    padding-right: 0;
+  } @else {
+    padding-right: #{$size_in_px}px;
+    padding-right: #{get-rem-from-px($size_in_px)}rem;
+  }
+}
 
 @mixin padding($size_in_px, $dimension: "") {
-  @if $dimension == top or $dimension == right or $dimension == bottom or $dimension == left {
+  @if $dimension == "" {
     @if $size_in_px == 0 {
-      padding-#{$dimension}: 0;
+      padding: 0;
     } @else {
-      padding-#{$dimension}: #{$size_in_px}px;
-      padding-#{$dimension}: #{get-rem-from-px($size_in_px)}rem;
+      padding: #{$size_in_px}px;
+      padding: #{get-rem-from-px($size_in_px)}rem;
     }
   }
-  @else if $dimension == "" and type_of($size_in_px) == list {
-    $parts-px: ();
-    $parts-rem: ();
-    @each $size in $size_in_px {
-      @if $size == 0 {
-        $parts-px: append($parts-px, 0, "space");
+  @if $dimension == "inline" {
+    @if $size_in_px == 0 {
+      padding-left: 0;
+      padding-right: 0;
+      padding-inline: 0;
+    } @else if type_of($size_in_px) == list and length($size_in_px) >= 2 {
+
+      $firstValue: nth($size_in_px, 1);
+      $secondValue: nth($size_in_px, 2);
+
+      @if $firstValue == $secondValue {
+
+        @include _padding-left($firstValue);
+        @include _padding-right($secondValue);
+
       } @else {
-        $parts-px: append($parts-px, #{$size}px, "space");
+
+        &[dir="ltr"] {
+          @include _padding-left($firstValue);
+          @include _padding-right($secondValue);
+        }
+
+        &[dir="rtl"] {
+          @include _padding-right($firstValue);
+          @include _padding-left($secondValue);
+        }
+
       }
-    }
-    @each $size in $size_in_px {
-      @if $size == 0 {
-        $parts-rem: append($parts-rem, 0, "space");
+
+      $derivedFirstValue: 0;
+      $derivedSecondValue: 0;
+      @if $firstValue != 0 {
+        $derivedFirstValue: #{get-rem-from-px($firstValue)}rem;
+      }
+      @if $secondValue != 0 {
+        $derivedSecondValue: #{get-rem-from-px($secondValue)}rem;
+      }
+      @if $firstValue == $secondValue {
+        padding-inline: $derivedFirstValue;
       } @else {
-        $parts-rem: append($parts-rem, #{get-rem-from-px($size)}rem, "space");
+        &#{$libero-selector-to-force-sass-output-order} {
+          padding-inline: $derivedFirstValue $derivedSecondValue;
+        }
       }
+
+    } @else {
+      @include _padding-left($size_in_px);
+      @include _padding-right($size_in_px);
+      padding-inline: #{get-rem-from-px($size_in_px)}rem;
     }
-    padding: $parts-px;
-    padding: $parts-rem;
   }
-  @else {
+
+  @if $dimension == "inline-start" {
     @if $size_in_px == 0 {
-      padding: 0;
+      &[dir="ltr"] {
+        padding-left: 0;
+      }
+      &[dir="rtl"] {
+        padding-right: 0;
+      }
+      &#{$libero-selector-to-force-sass-output-order} {
```

Comment:
If we're mandating `<html dir="...">`, could use `[dir] &`?

Reply:
Yes. It means we can lose the hack! 🎉 

I think `html[dir] &` reads better in the context . So this would now generate, for example: 
```
html[dir="ltr"] .foo:not([dir]),
.foo[dir="ltr"] {
  padding-left: 400px;
  padding-left: 25rem;
  padding-right: 800px;
  padding-right: 50rem; }

html[dir="rtl"] .foo:not([dir]),
.foo[dir="rtl"] {
  padding-right: 400px;
  padding-right: 25rem;
  padding-left: 800px;
  padding-left: 50rem; }

html[dir] .test {
  padding-inline: 25rem 50rem; }
```